specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json

```diff
@@ -1098,34 +1107,111 @@
       },
       "description": "Class representing a Traffic Manager HeatMap traffic flow properties."
     },
-    "HeatMapProperties": {
+    "HeatMapPropertyBase": {
+      "type": "object",
+      "discriminator": "heatMapType",
       "properties": {
-        "startTime": {
-          "type": "string",
-          "format": "date-time",
-          "description": "The beginning of the time window for this HeatMap, inclusive."
+        "heatMapType": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "heatMapType"
+      ],
+    },
+    "default":{
+      "description": "Sub-class representing Traffic Manager HeatMap default properties.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/HeatMapPropertyBase",
+          "type": "object"
         },
-        "endTime": {
-          "type": "string",
-          "format": "date-time",
-          "description": "The ending of the time window for this HeatMap, exclusive."
+        {
+          "properties": {
+            "startTime": {
+              "type": "string",
+              "format": "date-time",
+              "description": "The beginning of the time window for this HeatMap, inclusive."
+            },
+            "endTime": {
+              "type": "string",
+              "format": "date-time",
+              "description": "The ending of the time window for this HeatMap, exclusive."
+            },
+            "endpoints": {
+              "type": "array",
+              "description": "The endpoints used in this HeatMap calculation.",
+              "items": {
+                "$ref": "#/definitions/HeatMapEndpoint"
+              }
+            },
+            "trafficFlows": {
+              "type": "array",
+              "description": "The traffic flows produced in this HeatMap calculation.",
+              "items": {
+                "$ref": "#/definitions/TrafficFlow"
+              }
+            }
+          }
+        }
+      ]
+    },
+    "asn":{
+      "description": "Sub-class representing Traffic Manager HeatMap ASN location properties.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/HeatMapPropertyBase",
+          "type": "object"
         },
-        "endpoints": {
-          "type": "array",
-          "description": "The endpoints used in this HeatMap calculation.",
-          "items": {
-            "$ref": "#/definitions/HeatMapEndpoint"
+        {
+          "properties": {
+            "trafficFlows": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/TrafficLocation"
+              }
+            }
           }
+        }
+      ]
+    },
+    "country":{
```

Comment:
Use a meaingful name, and the `x-ms-discriminator-value` for the real value. Example:
```json
    "DatasetBZip2Compression": {
      "x-ms-discriminator-value": "BZip2",
```
This means "Create class DatasetBZip2Compression, but match value BZip2 on the wire"